source/frontend/src/hooks/useApiUserOverride.ts

```diff
@@ -94,6 +98,23 @@ export const useApiUserOverride = <
         okButtonVariant: 'warning',
         cancelButtonText: 'Cancel Update',
       });
+    } else if (
+      state?.userOverrideCode &&
+      state.userOverrideCode === UserOverrideCode.CONTRACTOR_SELFREMOVED
+    ) {
+      setModalContent({
+        title: 'Note',
+        message: RemoveSelfContractorContent(),
+        handleOk: async () => {
+          setState({
+            previousUserOverrideCodes: [...state.previousUserOverrideCodes],
+            userOverrideCode: null,
+            message: null,
+          });
+          setDisplayModal(false);
+        },
+        okButtonText: 'Close',
+      });
```

Comment:
Not sure how I feel about this change because the pattern for using this hook is to ask user for confirmation before retrying the same action, whereas the "contractor removed" popup does not retry, it just tells you that you cannot do that. Perhaps we could put this logic in a separate hook for the use cases where we just want to display a popup without a retry?

What are your thoughts @devinleighsmith and @FuriousLlama ?

Reply:
this hook is hoarding the modal context and the catch errors. Errors catch outside this hook won't display the Override confirmation and modal context outside this hook won't work.

Reply:
I understand what you are saying so perhaps we will discuss this on our improvement meeting and log a tech-debt ticket to improve in the future, thanks for explaining